examples/core/Graphormer/dataset.py

```diff
@@ -0,0 +1,107 @@
+"""
+This file contains the MolHIVDataset class, which handles data preprocessing
+(computing required graph features, converting graphs to tensors) of the
+ogbg-molhiv dataset.
+"""
+import torch as th
+import torch.nn.functional as F
+from dgl import shortest_dist
+from ogb.graphproppred import DglGraphPropPredDataset
+from torch.nn.utils.rnn import pad_sequence
+
+
+class MolHIVDataset(th.utils.data.Dataset):
+    def __init__(self):
+        dataset = DglGraphPropPredDataset(name="ogbg-molhiv")
+        split_idx = dataset.get_idx_split()
+
+        # compute the shortest path distance during preprocessing
+        for g, label in dataset:
+            spd, path = shortest_dist(g, root=None, return_paths=True)
+            g.ndata["spd"] = spd
+            g.ndata["path"] = path
+
+        self.train, self.val, self.test = (
+            dataset[split_idx["train"]],
+            dataset[split_idx["valid"]],
+            dataset[split_idx["test"]],
+        )
+
+    def collate(self, samples):
+        # To be consistent with the input style of Graphormer, all graph
+        # features need to be padded into the same size (different number
+        # of nodes may make them inconsistent, and we align them with max
+        # number of nodes).
+        graphs, labels = map(list, zip(*samples))
+        labels = th.stack(labels)
+
+        num_graphs = len(graphs)
+        num_nodes = [g.num_nodes() for g in graphs]
+        max_num_nodes = max(num_nodes)
+
+        # +1 for the virtual node
```

Comment:
Worth to explain what's virtual node somewhere.
```suggestion
        # +1 for the virtual node.
```